lib/SecretsCrypto/sf-secrets-manager.c

```diff
@@ -153,27 +182,66 @@ static gboolean _sf_secrets_manager_perform_interaction(SfSecretsManager *manage
     SfSecretsManagerPrivate *priv = sf_secrets_manager_get_instance_private(manager);
     gchar *id = g_uuid_string_random();
     gboolean *result;
-    SfSecretsInteractionRequest *request = g_object_new(SF_TYPE_SECRETS_INTERACTION_REQUEST,
+    SfSecretsInteractionRequest *request;
+    const gchar *secret_name;
+    const gchar *collection_name;
+    const gchar *plugin_name;
+    const gchar *application_id;
+    gint32 operation;
+    const gchar *authentication_plugin_name;
+    GVariant *prompt_text;
+    GHashTable *pt;
+    gint32 input_type;
+    gint32 echo_mode;
+
+    g_variant_get(args,
+            "(&s&s&s&s(i)&s@a{is}(i)(i))",
+            &secret_name,
```

Comment:
This ig_variant_get() s all new function to me... so, I'm wondering that apparently there is no return value and possible deconstruction hiccups can't be detected that way -> Is there any guarantee that all these uninitialized pointers actually get cleared / filled in before they are used later on?

Reply:
Good point, it is simply returns without doing anything if the string doesn't match the value. If guess some additional checks would be smort to avoid undefined things.

Reply:
Well, initializing pointers to null would already make any (theoretical?) problems more likely to manifest themselves in deterministic manner / right context via immediate crash at null deref vs stack littering, or?

Reply:
Well, yes, that would be one way of addressing it. (Obviously the wrong one <flamebait emoji>). I would prefer to add a g_variant_check_format_string call (maybe adding a helper that does _check and _get).

Reply:
As far as I am concerned the only obviously wrong approach is to allow uninitialized pointers to be used as valid ones. The rest is subjective, depends on circumstances, and library of books have been written about the supposedly best ways to deal with it.

Reply:
Agreed.